### ISSUE TITLE
AdminV2 not using admin hostname

### ISSUE THREAD

**Issue by wisebaldone** [keycloak/keycloak#12747]
### Describe the bug

By setting the hostname and admin hostname you can log in fine but the client will attempt to reach out to admin endpoints on the main hostname which are suggested to be blocked.

Example request:

The admin client pings the whoami endpoint which is defined here: https://github.com/keycloak/keycloak-nodejs-admin-client/blob/main/src/resources/whoAmI.ts#L12 which uses (from my understanding) this https://github.com/keycloak/keycloak-admin-ui/blob/main/src/context/auth/AdminClient.tsx#L73 which is set via https://github.com/keycloak/keycloak/blob/main/services/src/main/java/org/keycloak/services/resources/admin/AdminConsole.java#L337

### Version

18.0.2

### Expected behavior

The admin client should communicate to all admin endpoints using the admin base url.

### Actual behavior

The admin client communicates using the frontend urls.

### How to Reproduce?

To reproduce failure you need a level 7 firewall to block off the paths as defined here: https://www.keycloak.org/server/reverseproxy

### Anything else?

Related PRs: https://github.com/keycloak/keycloak-admin-ui/pull/2704

**Comment by luchsamapparat** [comment]
I ran into the same issue. Looking at the code it's pretty obvious that configuring another URL for the admin console via `KC_HOSTNAME_ADMIN` (or in my case `KC_HOSTNAME_ADMIN_URL`) still requires `KC_HOSTNAME_URL` to also be accessible, as long as `@keycloak/keycloak-admin-client` uses the same URL for both authentication and admin requests.

```
export async function initAdminClient() {
  const keycloak = new Keycloak({
    url: environment.authServerUrl,👈 authServerUrl used as authentication URL
    realm: environment.loginRealm,
    clientId: environment.isRunningAsTheme
      ? "security-admin-console"
      : "security-admin-console-v2",
  });

  await keycloak.init({ onLoad: "check-sso", pkceMethod: "S256" });

  const adminClient = new KeycloakAdminClient();

  adminClient.setConfig({ realmName: environment.loginRealm });
  adminClient.baseUrl = keycloak.authServerUrl ?? environment.authServerUrl; //  👈 authServerUrl used as base URL for admin requests
  adminClient.registerTokenProvider({
    async getAccessToken() {
      try {
        await keycloak.updateToken(5);
      } catch (error) {
        keycloak.login();
      }

      return keycloak.token;
    },
  });

  return { keycloak, adminClient };
}
```

**Comment by thomasdarimont** [comment]
I just tried to verify this with the latest Keycloak nightly build:
with `keycloak.local` and `admin.local` point to 127.0.0.1 in `/etc/hosts`
```
$ docker run --rm -it \
    -p 8080:8080 \
    -e KEYCLOAK_ADMIN=admin \
    -e KEYCLOAK_ADMIN_PASSWORD=admin \
    quay.io/keycloak/keycloak:nightly \
    start-dev \
   --http-enabled=true \
   --http-relative-path=auth \
   --hostname-strict=true \ 
   --hostname-strict-https=false \
   --hostname=keycloak.local \
   --hostname-admin=admin.local
```

I can access the the admin console via `http://admin.local:8080/auth/admin`  via login at `http://keycloak.local:8080/auth/admin`
The admin-console v2 uses `http://keycloak.local:8080/auth` as backend url for API requests, e.g to `http://keycloak.local:8080/auth/admin/realms` . This should rather be `http://admin.local:8080/auth/admin`.
The account console is also accessible via `http://keycloak.local:8080/auth/realms/master/account`. 

Besides that, I get no CORS errors or other problems - admin-console is loading fine. I guess in your example you're not allowing admin endpoints to be served via the frontend URL which makes sense.

So in this case the admin-console Keycloak client should be configured to use `keycloak.local` als the URL for the login, but `admin.local` for API requests. I think we need to revise the test that validates this scenario.

**Comment by thomasdarimont** [comment]
@jonkoops would you mind taking a look at this? I think the PR https://github.com/keycloak/keycloak-ui/pull/3373 fixes this issue.
Is it somehow possible to create an additional tests setup for the e2e tests where the keycloak instance uses a frontend hostname (`--hostname`) != admin hostname (`--hostname-admin`) where all admin API requests to `/admin` to the frontend hostname are denied and admin API requests via admin hostname are allowed?

**Comment by jonkoops** [comment]
Yeah, this seems to be a pretty big issue that should be addressed quickly. @ssilvert could you move this issue to our repo?
